concordium-consensus/src/Concordium/External.hs

```diff
@@ -1135,17 +1135,22 @@ checkIfWeAreFinalizer cptr = do
 -- Returns -1 if we are not added as a baker.
 -- Returns -2 if we are added as a baker, but not part of the baking committee yet.
 -- Returns -3 if we have keys that do not match the baker's public keys on the chain.
+-- Returns -5 if baking has stopped due to consensus being shut down.
 -- Returns >= 0 if we are part of the baking committee. The return value is the
 -- baker id as appearing in blocks.
 bakerIdBestBlock :: StablePtr ConsensusRunner -> IO Int64
 bakerIdBestBlock cptr = do
     (ConsensusRunner mvr) <- deRefStablePtr cptr
-    res <- runMVR Q.getBakerStatusBestBlock mvr
-    return $! case res of
-        NoBaker -> -1
-        InactiveBaker _ -> -2
-        BadKeys _ -> -3
-        ActiveBaker bid -> fromIntegral bid
+    isDown <- runMVR Q.checkIsShutDown mvr
```

Comment:
It might be better to have a separate query for checking if the node is shutdown. Otherwise, extend the definition of `BakerStatus` and revise the implementation of `getBakerStatusBestBlock`. There should be minimal logic in this module, beyond what is needed to interface between Rust and Haskell.

Reply:
Makes sense. I'll revise as suggested.